### DISCUSSION TITLE
Sonoff ZIGBEE 3.0 USB DONGLE PLUS + Synology DS218+

### DISCUSSION THREAD

**Discussion by ghost** [indomus/forum#1355] [HUB: Home Assistant]
Ciao a tutti, premetto che sono nuovo del mondo della domotica ma "vecchio" nel mondo dell'informatica.
Premesso questo ho acquistato una Sonoff ZIGBEE 3.0 USB DONGLE PLUS per collegarla al mio NAS Synology DS218+.
Creato vm, installato ha ma non riesco a far funzionare Zigbee2mqtt.
Attualmente la configurazione di Zigbee2mqtt è questa:

```
data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  user: username_creato
  password: password
serial:
  port: /dev/ttyUSB0
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
availability: false
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false
```

Gli unici componenti aggiuntivi attivi sono:

- Mosquitto broker
- Termina & SSH
- Z-Wave JS
- Zigbee2mqtt

Il sistema si avvia:

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[21:12:24] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[21:12:25] INFO: MQTT available, fetching server detail ...
[21:12:25] INFO: MQTT server settings not configured, trying to auto-discovering ...
[21:12:26] INFO: Configuring 'mqtt://core-mosquitto:1883' mqtt server
[21:12:26] INFO: Previous config file found, checking backup
[21:12:26] INFO: Creating backup config in '/config/zigbee2mqtt/.configuration.yaml.bk'
[21:12:26] INFO: Adjusting Zigbee2mqtt core yaml config with add-on quirks ...
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[21:12:27] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.2 start
> node index.js
```

Fino a qui tutto ok, da qui sotto partono gli errori

```
Zigbee2MQTT:error 2022-01-20 21:12:56: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-01-20 21:12:56: Failed to start zigbee
Zigbee2MQTT:error 2022-01-20 21:12:56: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-01-20 21:12:56: Exiting...
Zigbee2MQTT:error 2022-01-20 21:12:56: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:102:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:123:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:100:27)
    at start (/app/index.js:97:5)
```

Ho provato a seguire diverse guide e tutorial ma non ho trovato alcuna soluzione.. per caso c'è qualcuno di voi che ha lo stesso errore?

**Comment by snakuzzo** [comment]
Devi innanzi tutto installare i driver USB Serial come abbiamo indicato nella guida di zigbee2mqtt su Docker su QNAP.

Poi...

- Se usi Docker, hai semplicemente bisogno di mappare la porta dell'host con quella del container
- Se invece usi una VM devi abilitare la porta nelle impostazioni della VM

Solo dopo puoi provare ad avviare zigbee2mqtt.

**Reply by ghost** [reply]
Dopo faccio una prova ma attualmente mi è sorto un dubbio... volevo prima aggiornare il firmware tramite SmartRF usando il
`CC1352P2_CC2652P_launchpad_coordinator_20211217.hex`
Una volta inserite la usb sia in "boot" che normale NON viene rilevata dal software SmartRF... ho provato diverse USB del mio pc e mi sembra impossibile che non venga rilevata... potrebbe essere guasta.

**Reply by snakuzzo** [reply]
Io provato su Linux e su Raspberry (fatta procedura di flash Linux ovviamente con lo script python) e non ho avuto alcun problema.
La chiavetta sonoff basta inserirla e flasharla

**Reply by snakuzzo** [reply]
Ma comunque il problema non è il firmware. Il problema che hai è che sul QNAP devi installare i driver terze parti USB Serial e poi devi montare la chiavetta nelle impostazioni della VM

**Reply by ghost** [reply]
Scusa ma sono stato via questo weekend e non ho potuto fare dei test.
Allora, premetto che io ho Synology e non QNAP ma credo che poco cambi.
Sono riuscito ad aggiorare il firmware zigbee2mqtt installando correttamente il 

`CC1352P2_CC2652P_launchpad_coordinator_20211217`

Ho scaricato ed installato i drivers usb del mio DS218+ chip ApolloLake in
`/lib/modules`

ed avviato come da guida
https://community.synology.com/enu/forum/1/post/144408

Ho inoltre montato il dispositivo nella VM ma quello l'ho sempre abilitato già in precedenza.

Ed ora mi da un errore diverso:

`[21:41:45] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.2 start
> node index.js
Zigbee2MQTT:error 2022-01-24 21:41:53: Configuration is not consistent with adapter state/backup!
Zigbee2MQTT:error 2022-01-24 21:41:53: - PAN ID: configured=6754, adapter=6754
Zigbee2MQTT:error 2022-01-24 21:41:53: - Extended PAN ID: configured=00124b0024c69c8b, adapter=00124b0024c69c8b
Zigbee2MQTT:error 2022-01-24 21:41:53: - Network Key: configured=01030507090b0d0f00020406080a0c0d, adapter=01030507090b0d0f00020406080a0c0d
Zigbee2MQTT:error 2022-01-24 21:41:53: - Channel List: configured=11, adapter=11
Zigbee2MQTT:error 2022-01-24 21:41:53: Please update configuration to prevent further issues.
Zigbee2MQTT:error 2022-01-24 21:41:53: If you wish to re-commission your network, please remove coordinator backup at /config/zigbee2mqtt/coordinator_backup.json.
Zigbee2MQTT:error 2022-01-24 21:41:53: Re-commissioning your network will require re-pairing of all devices!
Zigbee2MQTT:error 2022-01-24 21:41:53: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-01-24 21:41:53: Failed to start zigbee
Zigbee2MQTT:error 2022-01-24 21:41:53: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-01-24 21:41:53: Exiting...
Zigbee2MQTT:error 2022-01-24 21:41:53: Error: startup failed - configuration-adapter mismatch - see logs above for more information
    at ZnpAdapterManager.determineStrategy (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:229:35)
    at ZnpAdapterManager.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:63:26)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:123:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:100:27)
    at start (/app/index.js:97:5)`

sta usb mi stà tirando scemo..

**Reply by snakuzzo** [reply]
hai letto cosa dice il log ?

`Configuration is not consistent with adapter state/backup!`

Quindi qui hai modificato qualcosa rispetto al backup che avevi che rende il tutto inconsistente

e poi qui...

```
Please update configuration to prevent further issues.
If you wish to re-commission your network, please remove coordinator backup at /config/zigbee2mqtt/coordinator_backup.json.
```

e sotto, quando fallice, ti dice di guardare appunto gli errori sopra...

```
Error: startup failed - configuration-adapter mismatch - see logs above for more information
```

Per quanto riguarda Synology...non ce l'ho e non so aiutarti più di tanto. So solo che con DSM7 non si potevano più usare periferiche USB.

**Comment by ghost** [comment]
Ok sono riuscito a farlo funzionare. Sostanzialmente avendo aggiornato il firmware non matchava con i dati salvati del firmware precedente.
Ora funziona tutto, grazie :)

**Reply by snakuzzo** [reply]
bene. Ma quindi non sei su DSM7 ?

**Reply by ghost** [reply]
No al contrario sono su DSM 7.0.1-42218 Update 2

**Reply by snakuzzo** [reply]
Per caso usi Telegram?
Ci sono alcune persone nella nostra chat che hanno synology con dsm7 e volevano capire meglio questa cosa

**Reply by ghost** [reply]
Ciao, si lo uso poco ma lo uso in chan vi trovo?

**Reply by snakuzzo** [reply]
Si ci trovi lì... https://t.me/indomus_chat
